src/common/events.ts

```diff
@@ -1,4 +1,25 @@
-export interface SaveFileEventPayload {
+export const enum EventType {
+  SaveFileRequest,
+  SaveFileResponse,
+  CompileModelRequest,
+  CompileModelResponse,
+}
+
+export interface EventPayload {
+  type: EventType;
+  errorMessage?: string;
+}
+
+export interface SaveFileRequestPayload extends EventPayload {
+  name: string;
+  file: File;
+}
+
+export interface CompileModelRequestPayload extends EventPayload {
   name: string;
   file: File;
 }
+
+export interface CompileModelResponsePayload extends EventPayload {
+  code?: string;
+}
```

Comment:
I would use a [discriminating union](https://www.typescriptlang.org/docs/handbook/unions-and-intersections.html#discriminating-unions) here instead.

Why?

 1. use a **self-documenting string** as the `type` property that is visible while debugging!
 2. TypeScript's type inferencing algorithm understands discriminating unions when you use them in a **switch** or **if**. That means, **no more `if (event.type === BLAH) { payload as BlahPayload }`. TypeScript already _infers_ that it's a `BlahPayload`!


```suggestion
export type AppWorkerEvent =
  | SaveFileRequest
  | CompileModelRequest
  | CompileModelResponse;

interface EventPayload {
  errorMessage?: string;
}

export interface SaveFileRequest extends EventPayload {
  type: "save-file";
  name: string;
  file: File;
}

export interface CompileModelRequest extends EventPayload {
  type: "compile-model";
  name: string;
  file: File;
}

export interface CompileModelResponse extends EventPayload {
  type: "compilation-result";
  code?: string;
}
```